lib/pure/net.nim

```diff
@@ -2054,3 +2054,29 @@ proc getPrimaryIPAddr*(dest = parseIpAddress("8.8.8.8")): IpAddress =
     result = socket.getLocalAddr()[0].parseIpAddress()
   finally:
     socket.close()
+
+func toIpAddress*(ip: int64): IpAddress {.since: (1, 5).} =
+  ## Convert an `int64` to an `IpAddress` of family `IPv4`.
+  assert ip >= 0, "IP address must be a positive integer"
+  result = IpAddress(family: IpAddressFamily.IPv4, address_v4: [
+    uint8(ip div 16_777_216'i64 mod 256'i64),
+    uint8(ip div 65_536'i64 mod 256'i64),
+    uint8(ip div 256'i64 mod 256'i64),
+    uint8(ip mod 256'i64),
```

Comment:
let ip = cast[uint64](ip) 
+ shr stuffs
?

Reply:
Yeah, use bitmasks and shifts here.